fenix/app/src/main/java/org/mozilla/fenix/compose/SwipeToDismiss.kt

```diff
@@ -0,0 +1,93 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.gestures.Orientation
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.offset
+import androidx.compose.material.DismissDirection
+import androidx.compose.material.DismissDirection.EndToStart
+import androidx.compose.material.DismissDirection.StartToEnd
+import androidx.compose.material.DismissState
+import androidx.compose.material.DismissValue
+import androidx.compose.material.DismissValue.Default
+import androidx.compose.material.DismissValue.DismissedToEnd
+import androidx.compose.material.DismissValue.DismissedToStart
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.FractionalThreshold
+import androidx.compose.material.ThresholdConfig
+import androidx.compose.material.swipeable
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+import kotlin.math.roundToInt
+
+/**
+ * A composable that can be dismissed by swiping left or right
+ *
+ * @param state The state of this component.
+ * @param modifier Optional [Modifier] for this component.
+ * @param enabled [Boolean] controlling whether the content is swipeable or not.
+ * @param directions The set of directions in which the component can be dismissed.
+ * @param dismissThreshold The threshold the item needs to be swiped in order to be dismissed.
+ * @param background A composable that is stacked behind the content and is exposed when the
+ * content is swiped. You can/should use the [state] to have different backgrounds on each side.
+ * @param dismissContent The content that can be dismissed.
+ */
+@Composable
+@ExperimentalMaterialApi
+fun SwipeToDismiss(
+    state: DismissState,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    directions: Set<DismissDirection> = setOf(EndToStart, StartToEnd),
+    dismissThreshold: ThresholdConfig = FractionalThreshold(DISMISS_THRESHOLD),
+    background: @Composable RowScope.() -> Unit,
+    dismissContent: @Composable RowScope.() -> Unit,
+) {
+    val swipeWidth = with(LocalDensity.current) {
+        LocalConfiguration.current.screenWidthDp.dp.toPx()
+    }
+
+    val anchors = mutableMapOf(0f to Default)
+    if (StartToEnd in directions) anchors += swipeWidth to DismissedToEnd
+    if (EndToStart in directions) anchors += -swipeWidth to DismissedToStart
+
+    val thresholds = { _: DismissValue, _: DismissValue ->
+        dismissThreshold
+    }
```

Comment:
nit: group the vals together then do the anchor maths
```suggestion
    val swipeWidth = with(LocalDensity.current) {
        LocalConfiguration.current.screenWidthDp.dp.toPx()
    }
    val anchors = mutableMapOf(0f to Default)
    val thresholds = { _: DismissValue, _: DismissValue ->
        dismissThreshold
    }
    
    if (StartToEnd in directions) anchors += swipeWidth to DismissedToEnd
    if (EndToStart in directions) anchors += -swipeWidth to DismissedToStart
```